packages/ifpack2/example/BlockTriDi.cpp

```diff
@@ -278,23 +356,81 @@ main (int argc, char* argv[])
   if(args.matrixFilename == "") {
     // matrix
     Teuchos::ParameterList plist;
-    plist.set("matrixType","Laplace1D");    
-    plist.set("nx", (GO)args.nx*comm->getSize());
+    if(args.matrixType == "") {
+      plist.set("matrixType","Laplace1D");
+    } else {
+      plist.set("matrixType", args.matrixType);
+      plist.set("nx", (GO)args.nx);
+      plist.set("mx", (GO)args.mx);
+      if(args.mx == 1 && args.ny == -1)
+        plist.set("mx", (GO)comm->getSize());
+      else if(args.mx != comm->getSize() && args.ny == -1) {
+        std::string msg = "mx is not consistent with comm->getSize";
+        throw std::runtime_error(msg);
+      }
+      else if(args.mx != 1) {
+        std::string msg = "mx != 1 is not yet supported";
+        throw std::runtime_error(msg);
+      }
+      if(args.ny > -1) {
+        plist.set("ny", (GO)args.ny);
+        plist.set("my", (GO)args.my);
+        if(args.mx*args.my == 1 && args.nz == -1)
+          plist.set("my", (GO)comm->getSize());
+        else if(args.mx*args.my != comm->getSize() && args.nz == -1) {
+          std::string msg = "mx and my are not consistent with comm->getSize";
+          throw std::runtime_error(msg);
+        }
+      }
+      if(args.nz > -1) {
+        plist.set("nz", (GO)args.nz);
+        plist.set("mz", (GO)args.mz);
+        if(args.mx*args.my*args.mz == 1)
+          plist.set("mz", (GO)comm->getSize());
+        else if(args.mx*args.my*args.mz != comm->getSize()) {
+          std::string msg = "mx, my, and mz are not consistent with comm->getSize";
+          throw std::runtime_error(msg);
+        }
+      }
+    }
+    if(0 < args.blockSize) {
+      plist.set("blockSize", args.blockSize);
+    }
     Ablock = BuildBlockMatrix<SC,LO,GO,NO>(plist,comm);
+    std::cout << "p=" << comm->getRank() << " | Ablock, local size: " << Ablock->getLocalNumRows() << std::endl;
 
     //rhs 
     B = rcp(new MV(Ablock->getRangeMap(),1));
     B->putScalar(Teuchos::ScalarTraits<SC>::one());
 
-    // line info (lpp lines per proc)
-    int line_length = std::max(1, args.nx  / args.lpp);
+    // line info (sublinesPerLine lines per proc along direction x)
+    if ( args.sublinesPerLine < 1 && args.sublinesPerLine != -1) {
+      std::string msg = "the value of sublinesPerLine = " + std::to_string(args.sublinesPerLine) + " is not supported";
+      throw std::runtime_error(msg);
+    }
+
+    int line_length = std::max(1, (int) std::ceil(args.nx  / args.sublinesPerLine));
+    int line_per_x_fiber = std::ceil(args.nx  / line_length);
```

Comment:
Please put a comment as to what an "x_fiber" is.

Reply:
Does something like this is clear enough?

```
    // We compute the number of lines oriented along the x direction of the mesh.
    // This number is called line_per_x_fiber where a fiber refers to an initial
    // x line in the mesh before dividing it in sublines.
```

Reply:
@kliegeois Yes, that explanation is clear, thank you.